### PR TITLE
Breaking: drop buffer support

### PR DESCRIPTION
**Pull request by jeswr**
- Drop support for `Buffer`s as input to `saveFileInContainer` and `overwriteFile`. Use the file name of inputs to `saveFileInContainer` as the `Slug` name recommendation if no slug is explicitly provided.

### CODE REVIEW

File: e2e/node/resource.test.ts [hunk 1]
```diff
@@ -146,28 +142,17 @@ describe("Authenticated end-to-end", () => {
...
-  it("can create, delete, and differentiate between RDF and non-RDF Resources using a node Buffer", async () => {
+  it("can create, delete, and differentiate between RDF and non-RDF Resources using a Blob from the node Buffer package", async () => {
     const fileUrl = `${sessionResource}.txt`;
 
     const sessionFile = await overwriteFile(
       fileUrl,
-      NodeBuffer.from("test"),
+      // We need to type cast because the buffer definition
+      // of Blob does not have the prototype property expected
+      // by the lib.dom.ts
```
Comment (NSeydoux) [inline e2e/node/resource.test.ts:152]: If I remember correctly, you had opened an issue about this a while ago. Do you happen to have the link to it?

File: e2e/node/resource.test.ts [hunk 2]
```diff
@@ -178,59 +163,32 @@ describe("Authenticated end-to-end", () => {
     await deleteFile(fileUrl, fetchOptions);
   });
 
-  it("can create, delete, and differentiate between RDF and non-RDF Resources using a Blob from the node Buffer package", async () => {
+  // Blob is only available globally Node 18 and above
```
Comment (NSeydoux) [inline e2e/node/resource.test.ts:166]:
```suggestion
```

File: e2e/node/resource.test.ts [hunk 3]
```diff
@@ -359,18 +317,15 @@ describe("Authenticated end-to-end", () => {
...
+      const response = await fetchOptions.fetch(args[0], {
+        ...args[1],
         headers: {
-          ...init?.headers,
+          ...args[1]?.headers,
           "User-Agent": "some-user-agent",
         },
       });
-      if (info.toString() === sessionResource) {
+      if (args[0].toString() === sessionResource) {
```
Comment (NSeydoux) [inline e2e/node/resource.test.ts:328]: Why this change? I find the previous version to be more readable.

File: src/resource/file.ts [hunk 4]
```diff
@@ -19,9 +19,9 @@
 // SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
 
-import type { Buffer, File as NodeFile } from "buffer";
+import type { File as NodeFile } from "buffer";
 import type {
-  File,
+  File as BlobFile,
```
Comment (NSeydoux) [inline src/resource/file.ts:24]: We have to remove that alias eventually, it's really confusing ^^. When Node 18 goes away, that's going with it.

File: src/resource/file.ts [hunk 5]
```diff
@@ -200,23 +201,8 @@ type SaveFileOptions = WriteFileOptions & {
...
-export async function saveFileInContainer<
-  FileExt extends File | NodeFile | Buffer,
->(
-  folderUrl: Url | UrlString,
-  file: FileExt,
-  options?: Partial<SaveFileOptions>,
-): Promise<FileExt & WithResourceInfo>;
 export async function saveFileInContainer<
-  FileExt extends File | NodeFile | Buffer,
+  FileExt extends File | BlobFile | NodeFile,
```
Comment (NSeydoux) [inline src/resource/file.ts:205]: What is the first `File` here?

Comment (jeswr) [reply src/resource/file.ts:205]: That's the one globally defined by typescript

Comment (jeswr) [reply src/resource/file.ts:205]: Even though the constructor doesn't work till mode 20, the type seems to exist in node 18

File: e2e/node/resource.test.ts (hunk 1 above)
Comment (jeswr) [reply e2e/node/resource.test.ts:152]:
```suggestion
      // We need to type cast because the buffer definition
      // of Blob does not have the prototype property expected
      // by the lib.dom.ts
      // See https://github.com/microsoft/TypeScript/issues/53668
      // and https://github.com/microsoft/TypeScript/issues/52166
```

@NSeydoux could you leave a comment in https://github.com/microsoft/TypeScript/issues/53668 to see if we can get this re-opened?

**Review by NSeydoux — APPROVED** [review]

File: e2e/node/resource.test.ts (hunk 3 above)
Comment (jeswr) [reply e2e/node/resource.test.ts]: My editor was complaining (see below) but this doesn't actually seem to be an issue in CI so I've reverted the changes.

image
